index.js

```diff
@@ -116,14 +117,10 @@ exports.init = function init(sbot) {
         const matchesQuery = QL0.toOperator(queryQL0)
         return cat([
           // Old
-          pull(
-            sbot.db.query(
-              where(and(matchesQuery, gt(latestSequence, 'sequence'))),
-              paginate(50),
-              toPullStream()
-            ),
-            pull.map(pull.values),
-            pull.flatten()
+          sbot.db.query(
+            where(and(matchesQuery, gt(latestSequence, 'sequence'))),
+            batch(75),
```

Comment:
Could you briefly explain the difference between `paginate` and `batch` in this context, as well as why we return `cat([sbot..db.query(...` and not `cat([pull(sbot.dq.query(...`?

Reply:
Ah right, sorry

`paginate` makes the pull-stream (at `toPullStream()`) deliver **arrays** of messages, so we had to use `pull.map(pull.values), pull.flatten()` to convert a `PullStream<Array<Msg>>` to a `PullStream<Msg>`.

`batch` is a new JITDB operator that makes the pull-stream (at `toPullStream()`) deliver messages simply, so it's already `PullStream<Msg>` and there's no flattening needed. But internally it has the same performance benefits of pagination (it "fetches" 75 at a time but delivers them one by one instead of 75 at once). See this PR that introduces batch: https://github.com/ssb-ngi-pointer/jitdb/pull/170

The surrounding `pull(  ... )` is just a way of doing piping. Now that we removed `pull.map(pull.values), pull.flatten()`, we would end up with only argument `x` passed to `pull( x )`, which is functionally the same as `x`, so that's why I got rid of the `pull( ... )`, where `x` is ` sbot.db.query( .... )`

Reply:
Awesome, thanks for the clear explanation!